### PR TITLE
board: MIMXRT1062-FMURT6: Add pinmux support for uart ports used for gps and telemtry.

### LINKED ISSUE #59005: Incorrect pinmux settings for MIMXRT1062_FMURT6 board
**Issue and Fix Description**
Internal pull up is required to make LPUART7 RX line (used as Console UART input) to work.
This is required since a reverse diode is connected to the LPUART7_RX line on the board. 

**To Reproduce**
1. west build -p always -b mimxrt1062_fmurt6 samples/net/eth_native_posix/
2. west flash
3. Unable to type any command on UART console

**Expected behavior**
User should be able to use Zephyr shell

### LINKED ISSUE #59010: ENET2 Module not being clocked on MIMXRT1062_FMURT6
**Issue Description**
50MHz Ref clock was not getting generated for ENET2 module.
Direction of the clock was from PHY to MAC

**To Reproduce**
1) west build -p always -b mimxrt1062_fmurt6 samples/net/eth_native_posix/
2) west flash
3) Carrier is always down, LINK LED is not up when connected to compatible link partner

**Expected behavior**
Carrier Up, IP and MAC configured for the ethernet interface, Orange Link LED should be lighted up.

**Fix**
Fix the sequence of Enet2 ref clk enablement -
1) Set ENET2 ref clock to be generated by External OSC - Setting Bits - #14 ENET2_CLK_SEL in register IOMUXC_GPR_GPR1 (0x400A_C004)
2) ENET2 ref clock direction as output - Set bit #18 - ENET2_TX_CLK_DIR in register IOMUXC_GPR_GPR1 (0x400A_C004)
3) Set ENET2 ref clk frequency to 50MHz - Setting 01 to Bit 2 and 3 of  0x400D_80E0 (CCM_ANALOG_PLL_ENET)

### LINKED ISSUE #59013: Kernel Crash during high throughput traffic over 2 wire ethernet on MIMXRT1062-FMURT6
**Issue Description**
On running high frequency ping kernel crash is observed 1 out of 20 times.
uart:~$ E:  HARD FAULT 
E:   Fault escalation (see below)
E:  BUS FAULT 
E:   Precise data bus error
E:   BFAR Address: 0x1c900102
E: r0/a1:  0x9e659210  r1/a2:  0x00000000  r2/a3:  0xe000ef74
E: r3/a4:  0x1c900102 r12/ip:  0x402a8180 r14/lr:  0x20000b41
E:  xpsr:  0x01000000
E: Faulting instruction address (r15/pc): 0x20000608
E: >>> ZEPHYR FATAL ERROR 25: Unknown error on CPU 0
E: Current thread: 0x20203b90 (idle)
E: Halting system

**To Reproduce**
1. west build -p always -b mimxrt1062_fmurt6 samples/net/eth_native_posix/
2. west flash
3. uart:~$ net ping 192.0.2.2 -s 1400 -i 0.5 -c 10
PING 192.0.2.2
1424 bytes from 192.0.2.2 to 192.0.2.1: icmp_seq=1 ttl=64 time=3 ms
1424 bytes from 192.0.2.2 to 192.0.2.1: icmp_seq=2 ttl=64 time=7 ms
1424 bytes from 192.0.2.2 to 192.0.2.1: icmp_seq=3 ttl=64 time=2 ms
uart:~$ E:  HARD FAULT 
E:   Fault escalation (see below)
E:  BUS FAULT 
E:   Precise data bus error
E:   BFAR Address: 0x1c900102
E: r0/a1:  0x9e659210  r1/a2:  0x00000000  r2/a3:  0xe000ef74
E: r3/a4:  0x1c900102 r12/ip:  0x402a8180 r14/lr:  0x20000b41
E:  xpsr:  0x01000000
E: Faulting instruction address (r15/pc): 0x20000608
E: >>> ZEPHYR FATAL ERROR 25: Unknown error on CPU 0
E: Current thread: 0x20203b90 (idle)
E: Halting system

**Expected behavior**
ping working without crash

**Fix**
RAM size was not configured properly so not enough memory for eth0 rx ring buffers.

### PR DESCRIPTION
**Pull request by sumitbatra-nxp**
1) This patch configures the correct pinmux settings for -

     lpuart2, lpuart3, lpuart5 ports used for gps and telemetry modules.

2) It also adds software pull up for the UART console port lpuart7 to 

3) It fixes an issue in ref clock configuration for ETH2 module

4) It also increases the RAM size

Fixes #59005 
Fixes #59010 
Fixes #59013

### CODE REVIEW

File: boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts [hunk 1]
```diff
@@ -33,6 +33,9 @@
 		zephyr,console = &lpuart7;
 		zephyr,shell-uart = &lpuart7;
 		zephyr,canbus = &flexcan1;
+		zephyr,gps1 = &lpuart2;
```
Comment (danieldegrasse) [inline boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts:36]: Are these well known Zephyr chosen node names? I don't see them being used anywhere in the tree. I would rather see these node names be devicetree aliases if these UART connections are being used in a downstream project

File: drivers/ethernet/eth_mcux.c [hunk 2]
```diff
@@ -1059,6 +1059,12 @@ static void eth_mcux_init(const struct device *dev)
 			kENET_RxAccelProtoCheckEnabled;
 	}
 
+	/* Set ENET2 ref clock to be generated by External OSC,*/
+	/* direction as output and frequency to 50MHz */
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(enet2), okay)
+	IOMUXC_GPR->GPR1 |= (IOMUXC_GPR_GPR1_ENET2_CLK_SEL_MASK |
```
Comment (danieldegrasse) [inline drivers/ethernet/eth_mcux.c:1065]: This looks like it should be handled at the `soc_rt10xx.c` level, not within the driver. Also, is there an `IOMUXC` API present for this bit, like `kIOMUXC_GPR_ENET1TxClkOutputDir` for ENET1?

File: boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts (hunk 1 above)
Comment (sumitbatra-nxp) [reply boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts]: Changed from Chosen to Aliases as suggested

File: drivers/ethernet/eth_mcux.c (hunk 2 above)
Comment (sumitbatra-nxp) [reply drivers/ethernet/eth_mcux.c]: Thanks @danieldegrasse .. I have modified as per your recommendation.. please check now..

**Comment by sumitbatra-nxp** [comment]
@danieldegrasse .. can you please re-check.. 
@dleach02.. @mmahadevan108  can you please review

**Comment by bperseghetti** [comment]
@danieldegrasse we added the issues for the bug report fixs and linked them.

File: boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts [hunk 3]
```diff
@@ -20,12 +20,39 @@
 		led1 = &red_led;
 		watchdog0 = &wdog0;
 		sdhc0 = &usdhc1;
-		pwm-0 = &flexpwm2_pwm0;
 		gps1 = &lpuart2;
 		telem1 = &lpuart3;
 		telem4-gps2 = &lpuart5;
 	};
 
+	pwm_shell: pwm_shell {
```
Comment (danieldegrasse) [inline boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts:28]: the commit title `MIMXRT1062-FMURT6: Fix PWM device names.` Should be updated to something like the following:
`soc: arm: nxp_imx: fix PWM node names`

File: soc/arm/nxp_imx/rt/soc_rt10xx.c [hunk 4]
```diff
@@ -66,7 +66,12 @@ const clock_enet_pll_config_t ethPllConfig = {
 #else
 	.enableClkOutput25M = false,
 #endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(enet), okay)
```
Comment (danieldegrasse) [inline soc/arm/nxp_imx/rt/soc_rt10xx.c:69]: Commit message here needs to be to something like- `soc: rt10xx: fix the sequence of Enet2 ref clk enablement`

File: boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts [hunk 5]
```diff
@@ -143,6 +142,7 @@
 	pinctrl-0 = <&pinmux_flexcan3>;
 	pinctrl-names = "default";
 	bus-speed = <125000>;
+	bus-speed-data = <1000000>;
```
[the remark this answers is not in the document]
Comment (bperseghetti) [reply boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts:145]: Done. Split it out into:
- board: MIMXRT1062-FMURT6: Set CANFD bus-speed-data
- board: MIMXRT1062-FMURT6: correct PWM prescaler

and renamed it to be more proper with:
- soc: nxp_rt10xx: add unique PWM names.

File: soc/arm/nxp_imx/rt/soc_rt10xx.c (hunk 4 above)
Comment (bperseghetti) [reply soc/arm/nxp_imx/rt/soc_rt10xx.c:69]: Commit title/message now updated.

**Review by danieldegrasse — APPROVED** [review]

**Comment by sumitbatra-nxp** [comment]
@mmahadevan108 .. Can you kindly review, as we need a second approver as well.

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**